client/go/cmd/clone.go

```diff
@@ -148,17 +135,50 @@ func getSampleAppsZip() (*os.File, error) {
 			return fmt.Errorf("could not download sample apps: %w", err)
 		}
 		defer response.Body.Close()
-		if response.StatusCode != 200 {
+		if response.StatusCode != http.StatusOK {
 			return fmt.Errorf("could not download sample apps: github returned status %d", response.StatusCode)
 		}
-		if err := f.Truncate(0); err != nil {
-			return fmt.Errorf("could not truncate sample apps file: %s: %w", f.Name(), err)
-		}
 		if _, err := io.Copy(f, response.Body); err != nil {
 			return fmt.Errorf("could not write sample apps to file: %s: %w", f.Name(), err)
 		}
+		if err := os.Rename(f.Name(), destination); err != nil {
+			return fmt.Errorf("could not move sample apps to cache path")
+		}
 		return nil
 	})
+}
+
+func getSampleAppsZip() (*os.File, error) {
+	cacheDir, err := vespaCliCacheDir()
+	if err != nil {
+		return nil, err
+	}
+	path := filepath.Join(cacheDir, sampleAppsFilename)
+	cacheExists := true
+	f, err := os.Open(path)
+	if errors.Is(err, os.ErrNotExist) {
+		cacheExists = false
+	} else if err != nil {
+		return nil, fmt.Errorf("could not open existing cache file: %w", err)
+	}
+	if cacheExists {
+		useCache, err := useCache(f)
+		if err != nil {
+			return nil, errHint(fmt.Errorf("could not determine cache status: %w", err), "Try ignoring the cache with the -f flag")
+		}
+		if useCache {
+			log.Print(color.Yellow("Using cached sample apps ..."))
+			return f, nil
+		}
+		f.Close()
+	}
+	f, err = os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0755)
```

Comment:
Instead of doing this, the method should simply return `os.Open(path)` at the end of this function (after the call to `fetchSampleAppsZip`)

Reply:
Simplified all of this, please take another look.